src/esp/assets/ResourceManager.cpp

```diff
@@ -2184,5 +2185,208 @@ std::unique_ptr<MeshData> ResourceManager::createJoinedCollisionMesh(
   return mesh;
 }
 
+void ResourceManager::getPrimitiveMeshData(const std::string& filename,
+                                           std::vector<float>& points,
+                                           std::vector<uint32_t>& triangles) {
+  // Get MeshData from createJoinedCollisionMesh
+  assets::MeshData::uptr joinedMesh = assets::MeshData::create_unique();
+  joinedMesh = createJoinedCollisionMesh(filename);
+
+  // Format into VHACD compatible vectors
+  for (vec3f point : joinedMesh->vbo) {
+    points.push_back(point[0]);
+    points.push_back(point[1]);
+    points.push_back(point[2]);
+  }
+
+  for (uint32_t index : joinedMesh->ibo) {
+    triangles.push_back(index);
+  }
+}
+
+bool ResourceManager::outputMeshMetaDataToObj(
+    const std::string& MeshMetaDataFile,
+    const std::string& new_filename,
+    const std::string& filepath) {
+  bool success = Cr::Utility::Directory::mkpath(filepath);
+
+  const MeshMetaData& metaData = getMeshMetaData(MeshMetaDataFile);
+  Cr::Utility::Directory::writeString(filepath + "/" + new_filename,
+                                      "# CHD Mesh group \n");
+  // write vertex info to file
+  int numVertices = 0;
+  for (MeshTransformNode node : metaData.root.children) {
+    CollisionMeshData& meshData =
+        meshes_.at(node.meshIDLocal + metaData.meshIndex.first)
+            ->getCollisionMeshData();
+    for (auto& pos : meshData.positions) {
+      Cr::Utility::Directory::appendString(filepath + "/" + new_filename,
```

Comment:
may want to consider a string build here

Reply:
Hm. I would actually fully build the string first and then write it at once. `appendString()` opens and closes the file each time and so it's way slower than it should be.

For the string building, [Utility::formatInto()](https://doc.magnum.graphics/corrade/namespaceCorrade_1_1Utility.html#aaf6e2f081fa8fd67065dd7b39811f5f6) is what you're looking for, the doc even has a snippet that is not too far from what you do here.